res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,860 @@
+    Prime4.rgbCode[deckColors[0] + "Dark"],
+    Prime4.rgbCode[deckColors[1] + "Dark"],
+    Prime4.rgbCode[deckColors[2] + "Dark"],
+    Prime4.rgbCode[deckColors[3] + "Dark"],
+];
+
+// Register '0x9n' as a button press and 'ox8n' as a button release
+components.Button.prototype.isPress = function(channel, control, value, status) {
+    return (status & 0xF0) === 0x90;
+};
+
+Prime4.DeckAssignButton = function(options) {
+
+    components.Button.call(this, options);
+
+    if (!Number.isInteger(this.deckIndex)) {
+        throw `invalid deckIndex: ${this.deckIndex}`;
+    }
+    if (!(this.toDeck instanceof Prime4.Deck)) {
+        throw "invalid toDeck";
+    }
+
+    const deckSide = (this.deckIndex % 2) === 0 ? "leftDeck" : "rightDeck";
+    if (!(Prime4[deckSide] instanceof Prime4.Deck)) {
+        throw "invalid deckIndex or structure; We expect Prime4.leftDeck and Prime4.rightDeck to be valid decks representing the physical left and right decks";
+    }
+    const isActive = function() {
+        return Prime4[deckSide] === this.toDeck;
+    };
+
+    this.trigger = function() {
+        this.output(isActive());
+    };
+
+    this.input = function(channel, control, value, status, _group) {
+        console.log(Object.keys(this));
+        if (!this.isPress(channel, control, value, status) || isActive()) {
+            return;
+        }
+        Prime4[deckSide].forEachComponent(c => { c.disconnect(); });
+        Prime4[deckSide] = this.decks[this.deckIndex];
+        this.assignmentButtons.forEachComponent(btn => btn.trigger());
+        Prime4[deckSide].forEachComponent(c => { c.connect(); c.trigger(); });
+    };
+};
+
+Prime4.DeckAssignButton.prototype = Object.create(components.Button.prototype);
+
+Prime4.EffectUnitEncoderInput = function(_channel, _control, value, _status, _group) {
+    if (value >= 1 && value < 20) {
+        this.inSetParameter(this.inGetParameter() + (value / 100));
+    } else if (value <= 127 && value > 100) {
+        this.inSetParameter(this.inGetParameter() + ((value - 128) / 100));
+    }
+};
+
+
+Prime4.init = function(_id, _debug) {
+    // Turn off all LEDs
+    midi.sendShortMsg(0x90, 0x75, 0x00);
+
+    // Initialise all three physical sections of the unit
+    /*
+    Prime4.leftDeck = new Prime4.Deck([1, 3], 4);
+    Prime4.rightDeck = new Prime4.Deck([2, 4], 5);
+    */
+
+    const decks = [
+        new Prime4.Deck(1, 4),
+        new Prime4.Deck(2, 5),
+        new Prime4.Deck(3, 4),
+        new Prime4.Deck(4, 5),
+    ];
+
+    // Bind the sections
+    Prime4.leftDeck = decks[0];
+    Prime4.rightDeck = decks[1];
+
+    Prime4.assignmentButtons = new components.ComponentContainer();
+    for (let i = 0; i < decks.length; i++) {
+        Prime4.assignmentButtons[i] = new Prime4.DeckAssignButton({
+            midi: [0x9F, 0x1C + i],
+            deckIndex: i,
+            toDeck: decks[i],
+            assignmentButtons: this.assignmentButtons,
+            decks: decks,
+        });
```

Comment:
this should fix the issue that the LEDs are only being lit after being pressed for the first time.
```suggestion
        });
        Prime4.assignmentButtons[i].trigger();
```